src/main/java/jenkins/plugins/git/GitUsernamePasswordBinding.java

```diff
@@ -0,0 +1,216 @@
+                                          EnvVars env, TaskListener listener) throws IOException, InterruptedException {
+        Git gitInstance = Git.with(listener, env).using(gitToolExe);
+        return gitInstance.getClient();
+    }
+
+    protected static final class GenerateGitScript extends AbstractOnDiskBinding<StandardUsernamePasswordCredentials> {
+
+        private final String userVariable;
+        private final String passVariable;
+        private final boolean unixNodeType;
+
+        protected GenerateGitScript(String gitUsername, String gitPassword,
+                                       String credentialId, boolean unixNodeType) {
+            super(gitUsername + ":" + gitPassword, credentialId);
+            this.userVariable = gitUsername;
+            this.passVariable = gitPassword;
+            this.unixNodeType = unixNodeType;
+        }
+
+        @Override
+        protected FilePath write(StandardUsernamePasswordCredentials credentials, FilePath workspace)
+                throws IOException, InterruptedException {
+            FilePath gitEcho;
+              //Hard Coded platform dependent newLine
+            if (this.unixNodeType) {
+                gitEcho = workspace.createTempFile("auth", ".sh");
+                // [#!/usr/bin/env sh] to be used if required, could have some corner cases
+                gitEcho.write("case $1 in\n"
+                        + "        Username*) echo " + this.userVariable
+                        + "                ;;\n"
+                        + "        Password*) echo " + this.passVariable
+                        + "                ;;\n"
+                        + "        esac\n", null);
+                gitEcho.chmod(0500);
+            } else {
+                gitEcho = workspace.createTempFile("auth", ".bat");
+                gitEcho.write("@ECHO OFF\r\n"
+                        + "SET ARG=%~1\r\n"
+                        + "IF %ARG:~0,8%==Username (ECHO " + this.userVariable + ")\r\n"
+                        + "IF %ARG:~0,8%==Password (ECHO " + this.passVariable + ")", null);
+            }
+            return gitEcho;
+        }
+
+        @Override
+        protected Class<StandardUsernamePasswordCredentials> type() {
+            return StandardUsernamePasswordCredentials.class;
+        }
+    }
+
+    @Symbol("GitUsernamePassword")
+    @Extension
+    public static final class DescriptorImpl extends BindingDescriptor<StandardUsernamePasswordCredentials> {
+
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return Messages.GitUsernamePasswordBinding_DisplayName();
+        }
+
+        public FormValidation doCheckGitToolName(@QueryParameter String toolName){
+            if(toolName == null){
+                return FormValidation.warning("Default Git Tool(Git/JGit) will be used");
+            }else {
+                return FormValidation.ok();
+            }
+        }
+
```

Comment:
I believe this code can be removed because there is no need to check for valid content when showing a pick list.  The pick list can only contain valid content.
```suggestion
```